book/src/building/swift-package/README.md

```diff
@@ -0,0 +1,214 @@
+# Bundling Rust code as a Swift package
+
+In this chapter we'll walk through bundling your Rust library into a Swift Package.
+
+Note that Swift Packages that contain native libraries only work on Apple hardware. 
+
+You should avoid bundling your Rust code into a Swift Package if you plan to target Linux, Windows or any other non-Apple target.
+
+## Project setup
+
+```bash
+mkdir rust-swift-package && cd rust-swift-package
+```
+
+### Rust project setup
+
+```bash
+cargo new my-rust-lib --lib
+cd my_rust_lib
+```
+
+```toml
+# my-rust-lib/Cargo.toml
+
+[lib]
+crate-type = ["staticlib"]
+
+[build-dependencies]
+swift-bridge-build = "0.1"
+
+[dependencies]
+swift-bridge = "0.1"
+```
+
+In `lib.rs`, add the following:
+
+```rust
+// my-rust-lib/src/lib.rs
+#[swift_bridge::bridge]
+mod ffi {
+    extern "Rust" {
+        fn hello_rust();
+    }
+}
+
+fn hello_rust() -> String {
+    String::from("Hello Rust!")
+}
+```
+
+Add a new `build.rs` file:
+```rust
+// my-rust-lib/build.rs
+
+use std::path::PathBuf;
+
+fn main() {
+    let out_dir = PathBuf::from("./generated");
+
+    let bridges = vec!["src/lib.rs"];
+    for path in &bridges {
+        println!("cargo:rerun-if-changed={}", path);
+    }
+
+    swift_bridge_build::parse_bridges(bridges)
+        .write_all_concatenated(out_dir, env!("CARGO_PKG_NAME"));
+}
+```
+
+Build the project for the desired platforms:
+
+```bash
+cargo build --target x86_64-apple-darwin
+cargo build --target aarch64-apple-ios
+cargo build --target x86_64-apple-ios
+```
+
+## Creating the XCFramework
+
+Go back to the root of the project and make a new directory.
+
+```bash
+mkdir MyFramework && cd $_
```

Comment:
Good idea! I'll change it this evening

Reply:
Awesome! Yeah ideally anyone going through our documentation can just copy and paste things. 

For example instead of only saying "create the file foo" we say something like 

Create the file foo.bar

`touch foo.bar`

---

Ok awesome. Thanks a lot for cleaning all of this up. Your work on Swift Package support is going to be really valuable for being able to expose Rust code to Swift.

Thanks again!

Reply:
I've learned a lot about Swift Packages from doing this, so it was a pleasure to figure this out.